src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -61,56 +77,23 @@ public RuntimeConfigProvider(RuntimeConfigLoader runtimeConfigLoader)
     /// <exception cref="DataApiBuilderException">Thrown when the loader is unable to load an instance of the config from its known location.</exception>
     public RuntimeConfig GetConfig()
     {
-        if (_runtimeConfig is not null)
+        if (ConfigLoader.RuntimeConfig is not null)
         {
-            return _runtimeConfig;
+            return ConfigLoader.RuntimeConfig;
         }
 
         // While loading the config file, replace all the environment variables with their values.
-        if (ConfigLoader.TryLoadKnownConfig(out RuntimeConfig? config, replaceEnvVar: true))
-        {
-            _runtimeConfig = config;
-            TrySetupConfigFileWatcher();
-        }
+        ConfigLoader.TryLoadKnownConfig(out _, replaceEnvVar: true);
 
-        if (_runtimeConfig is null)
+        if (ConfigLoader.RuntimeConfig is null)
```

Comment:
Can these two lines be combined? why can't you use the result of `TryLoadKnownConfig` ?